SudokuSolver/SudokuSolver/Common/Sudoku.cs

```diff
@@ -27,6 +27,53 @@ public Sudoku(int[,] grid)
             ValSqrt = (int)Math.Sqrt(RowsNumber);
         }
 
+        public int Get(int i, int j)
+        {
+            return Grid[i, j];
+        }
+
+        public void Set(int i, int j, int k)
+        {
+            Grid[i, j] = k;
+        }
+
+        public int Get((int i, int j) x)
+        {
+            return Grid[x.i, x.j];
+        }
+
+        public void Set((int i, int j) x, int k)
+        {
+            Grid[x.i, x.j] = k;
+        }
+
+        public IEnumerable<(int i, int j)> GetEmptyCells()
+        {
+            for(int i = 0; i < RowsNumber; ++i)
+            {
+                for (int j = 0; j < ColsNumber; ++j)
+                {
+                    if (Grid[i, j] == 0) yield return (i, j);
```

Comment:
Sam kiedyś robiłem takie returny/breaki/continue w tej samej linii co if, ale szczerze mówiąc lepiej to wygląda jeśli masz wcięcie. Jeśli tobie to nie przeszkadza to nie ma problemu